src/bsp/src/stm32/src/UWBRxFrame.cpp

```diff
@@ -0,0 +1,15 @@
+#include "UWBRxFrame.h"
+#include <cmath>
+
+float UWBRxFrame::getSFDAngle() const {
+    // RCPHASE register is only 7 bits, so mask off the highest bit
+    return (((float)(m_sfdAngleRegister & 0x7F)) / 64.0F) * M_PI;
+}
+
+float UWBRxFrame::getAccumulatorAngle() const {
+    // Accumulator data has a dummy byte at the first index
+    int16_t accumulatorQ = (m_firstPathAccumulator[2] << 8) | m_firstPathAccumulator[1];
```

Comment:
I think you could just cast it as a int16_t instead of using an or

`*((int16_t*) (m_firstPathaccumulator+1))`

Idk which one is more verbose on the intent or readable, but the cast does save you the trouble of shifting

Reply:
Changed for a cast for perfomance, but this makes the code non portable to big endian systems